text/0000-jsbox.md

```diff
@@ -111,45 +111,68 @@ fn get_user(mut cx: FunctionContext) -> JsResult<JsString> {
 }
 ```
 
-The instance of `Pool` can be returned to JavaScript using a `JsBox`. The `Pool` won't be dropped until the `JsBox` is garbage collected.
+The instance of `Pool` can be placed in a struct (`Db`) that implements `Finalize`. The `Db` struct can be returned to JavaScript using a `JsBox`. The instance of `Db` won't be dropped until the `JsBox` is garbage collected.
 
 ```rust
-fn create_pool(mut cx: FunctionContext) -> JsResult<JsBox<Pool>> {
-    let pool = Pool::new();
+struct Db {
+    pool: Pool,
+}
+
+impl Db {
+    fn new() -> Self {
+        Self { pool: Pool:new() }
+    }
+
+    fn pool(&self) -> &Pool {
+        &self.pool
+    }
 
-    Ok(cx.boxed(pool))
+    fn pool_mut(&mut self) -> &mut Pool {
+        &mut self.pool
+    }
+}
+
+// Use the default implementations
+impl Finalize for Db {}
+
+fn create_db(mut cx: FunctionContext) -> JsResult<JsBox<Db>> {
+    let db = Db::new();
+
+    Ok(cx.boxed(db))
 }
 ```
 
-The `Pool` instance can later be borrowed:
+The `Db` instance can later be borrowed:
 
 _Note: Through `std::ops::Deref` and the magic of auto-deref, `JsBox<T>` can be treated as `&T`._
 
 ```rust
 fn get_user(mut cx: FunctionContext) -> JsResult<JsString> {
-    let pool = cx.argument::<JsBox<Pool>>(0)?;
+    let db = cx.argument::<JsBox<Db>>(0)?;
     let id = cx.argument::<JsNumber>(1)?.value();
 
-    let username = pool.get_by_id(id)
+    let username = db.pool().get_by_id(id)
 
     Ok(cx.string(username))
 }
 ```
 
 Data may only be borrowed immutably. However, `RefCell` can be used to introduce interior mutability with dynamic borrow checking rules:
 
+## FIXME: Need to add a blanked impl of `Finalize` on common types like `RefCell`
```

Comment:
We should close this FIXME. If we aren't sure we've thought of all the std types we need blanket impls for, we can leave a note in the open questions section.